Add factory and resource rebuilding functionality to Rebuild Assets button

Land-Strider opened the pull request:
## What type of PR is this?
- [ ] Bug
- [ ] Change
- [ ] Feature
- [x] Miscellaneous

<details><summary><i><b>
Sub-categories:
</b></i></summary>

- [ ] Template
- [ ] Map
- [ ] Config
- [x] Function
- [ ] Localization

</details>

## What have you changed, and why?

**Information:**

> Added rebel factory and resource rebuilding functionality to the Commander Menu > Rebuild Assets button to allow rebels to regain income from destroyed economy markers without having to first give those markers to the enemy for repairs. Also added building repair functionality around factory/resource markers (for Building and House config classes).
Video showcase: https://youtu.be/e3q5oNAlUDg (income tick is set to 15 seconds for demo purposes only)

**How can your changes be tested, or reproduced?**

1. Get a rebel resource or factory marker DESTROYED status by killing its 4 workers, or capture a DESTROYED enemy marker. 
2. Confirm the destroyed marker status using the "Show Map Info" action on HQ map asset.
3. Use the Rebuild Assets button (Battle Menu > Commander > Commander Menu > Rebel HQ > Rebuild Assets)
4. Confirm the "Resource/factory near X town is rebuilt" successful task notification appears, 5000 money is spent, Show Map Info details for the marker no longer shows "DESTROYED" text under the garrison info, and the next resource tick has increased money income.

**Does this PR include changes not authored by you?**

- [X] No
- [ ] Yes (See below)

<details><summary><i><b>
Further info:
</b></i></summary><br>

- [ ] I confirm that I, and by extension this repository, can legally use these third-party changes. (Provide links or author attribution.)

> ...

</details>

## Please verify the following (If possible).

`*` - Mandatory

- [x] These changes are my own, or I have written permission to use them. `*`
- [x] These changes are ready for review, or will be marked as a draft. `*`
- [x] I have loaded the mission in LAN host.
- [x] I have loaded the mission on a dedicated server.

Is further work needed?

- [x] Needs further testing.
- [ ] Needs further changes.
- [ ] Needs to be converted to a draft.

## Please specify which Issue this PR Resolves (If Applicable).
This PR closes no active issue.

<hr><details><summary><i><b>
Notes:
</b></i></summary><br>

> ...

</details>

Land-Strider committed:
Add factory and resource rebuilding functionality into Commander Menu > Rebuild Assets button

UnseenKill commented on A3A/addons/core/functions/Base/fn_rebuildAssets.sqf:
```diff
@@ -30,12 +40,33 @@ switch (true) do {
+			[_x] remoteExec ["A3A_fnc_repairRuinedBuilding", 2]; // Repair each building
+		} forEach _buildings;
+
+		Debug_1("Calling A3A_fnc_rebuildEconomicAssets for %1", str _economyDead);
```
That looks a lot like what's happening in `A3A_fnc_rebuildEconomicAssets`, including the notification at L55.

Land-Strider replied:
Moved the building repair functionality to A3A_fnc_rebuildEconomicAssets completely. 

The notification at L55 is different than the one in A3A_fnc_rebuildEconomicAssets. This one is the Commander Menu screen notification below the map panel, the one in A3A_fnc_rebuildEconomicAssets at line 29 is Arma 3 task notification alike all-machines notification pop.

UnseenKill commented on A3A/addons/core/functions/Base/fn_rebuildAssets.sqf:
**General remarks:**

 * In `Debug_X` macro calls, the `str` is superfluous.
 * Maybe also revert whitespace-only changes in this file, please.

Land-Strider replied:
Removed the str as suggested. Also removed the whitespace only changes that my IDE didn't show by default.

UnseenKill commented on A3A/addons/core/functions/Base/fn_rebuildEconomicAssets.sqf:
```diff
@@ -0,0 +1,34 @@
+
+// Repair factory or resource buildings at the site
+private _economicSitePosition = getMarkerPos _economicSite;
+private _economicBuildings = nearestObjects [_economicSitePosition, A3A_buildingWhitelist, 500, true];
```
`A3A_buildingWhitelist` includes military buildings, only. This would be the right place for `_repairTypes` from changes in `A3A/addons/core/functions/Base/fn_rebuildAssets.sqf`.

Land-Strider replied:
Collected all building repair functionality near economic markers to A3A_fnc_rebuildEconomicAssets and made the incorrect array used there be switched with the _repairTypes as highlighted

Land-Strider committed:
Moved the redundant building repair function from fn_RebuildAssets.sqf to fn_RebuildEconomicAssets.sqf.
Removed the redundant str calls in debug and info statements.
Removed the unrelated whitespace changes.

UnseenKill commented on A3A/addons/core/functions/Base/fn_rebuildEconomicAssets.sqf:
```diff
@@ -0,0 +1,31 @@
+private _economicSitePosition = getMarkerPos _economicSite;
+private _economicBuildings = nearestObjects [_economicSitePosition, _repairTypes, 250];
+{
+    [_x] remoteExec ["A3A_fnc_repairRuinedBuilding", 2];
```
Sorry to be nitpicky here, but you're already in server context at this point

Land-Strider replied:
No problem, I'm trying learn as I go. Is it in the server context already because I called this file using remoteExec using 2 as the target in the fn_RebuildAssets.sqf?

UnseenKill replied:
Yes, also because you're checking for it earlier in line 5.

Land-Strider replied:
Oh right. I'll soon switch those remoteExecs in fn_rebuildEconomicAssets.sqf to call again.

Land-Strider replied:
Looks like the notification function call can't be done with call function as the UI drawing needs to be local on all machines I think. So only the repair function that is propagated to all machines via server anyway is reverted back to using call instead of remoteExec.

Land-Strider committed:
Changed the unnecesary remoteExec to a simple call, as the function is already running on the server.

UnseenKill reviewed (approved):
lgtm
